### PR TITLE
feat(express-tools): allow error handling for over ruling scopes

### LINKED ISSUE #36: [requireScopesMiddleware] is lacking error handling
https://github.com/zalando-incubator/lib-oauth-tooling/blob/master/src/express-tooling.ts#L40

```javascript
precedenceFunction(req, res, next)
.then(result => {
  if (result) {
    next();
  } else {
     validateScopes(req, res, next, scopes);
  }
});
```

If precendenceFunction throws an error this is not handled well.
There should be an `catch` clause that executes and error handler that is passed by from the calling function.
This handler would allow the application to answer with an 500 or 403 or whatever is intended.

I suggest to change the API:

```typescript
interface IPrecedenceFunction {
  (req: any, res: any, next: Function): Promise<boolean>;
}

interface IPrecedenceOptions {
  precedenceFunction: IPrecedenceFunction,
  precedenceErrorHandler: Function
}

function requireScopesMiddleware(scopes: string[],
                                 precedenceOptions?: IPrecedenceOptions) {
```

### PR DESCRIPTION
**Pull request by ISO50**
close #36

### CODE REVIEW

File: src/express-tooling.ts [hunk 1]
```diff
@@ -8,43 +8,67 @@ import {
...
           next();
         } else {
           validateScopes(req, res, next, scopes);
         }
+      })
+      .catch(err => {
+        try {
+          precedenceErrorHandler(err, logger);
+        } catch (err) {
+          logger.error('Error while exectuting precedenceErrorHandler: ', err);
```
Comment (Retro64) [inline src/express-tooling.ts]: Typo

**Comment by ISO50** [comment]
👍

**Comment by bzums** [comment]
The changes in `requireScopesMiddleware` should be documented in the readme right?, other then that it looks nice :)

**Comment by ISO50** [comment]
See #46

**Comment by bzums** [comment]
👍

**Comment by Retro64** [comment]
:+1:

**Pull request merged**